spec/index.bs

```diff
@@ -3765,17 +3765,18 @@ interface GPUQueue {
     void signal(GPUFence fence, GPUFenceValue signalValue);
 
     void writeBuffer(
-        ArrayBuffer sourceData,
-        GPUSize64 sourceOffset,
-        GPUBuffer destination,
-        GPUSize64 destinationOffset,
-        GPUSize64 size);
+        GPUBuffer buffer,
+        GPUSize64 bufferOffset,
+        ArrayBuffer data,
+        optional GPUSize64 dataOffset = 0,
+        optional GPUSize64 size = 0);
```

Comment:
Size should probably be optional with no default, so zero-size copies are really zero-size. In webgpu.h we can use WGPU_WHOLE_SIZE.

Reply:
In webgpu.h this wouldn't exist because the data is coming from a pointer. Done

Reply:
Ah right, it defaults to the size of the ArrayBuffer, not the size of the target buffer.